core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -289,13 +290,15 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
       val summary: NodeSeq =
         <div>
           <ul class="unstyled">
-            {if (parent.sc.isDefined) {
-              // Total duration is not meaningful unless the UI is live
-              <li>
-                <strong>Total Uptime: </strong>
-                {UIUtils.formatDuration(System.currentTimeMillis() - startTime)}
-              </li>
-            }}
+            <li>
+              <strong>Total Uptime:</strong>
+              {
+                if (endTime < 0) {
+                  val endTime = System.currentTimeMillis();
+                }
+                UIUtils.formatDuration(endTime - startTime)
```

Comment:
Hi @sarutak 
endTime is already set earlier in the AllJobPage. So it should be ok, no ? I think I'm missing something.
Please, can you explain what you mean be "alive" ?
Thanks a lot guys!
Regards
JB